### DISCUSSION TITLE
Mutability vs Immutability

### DISCUSSION THREAD

**Discussion by juanmaguitar** [WordPress/gutenberg#53018] [Interactivity API]
For developers used to Redux, or WordPress data, it could seem a bit weird not having a return statement in actions or effects. What is the reasoning behind this decision?

**Comment by DAreRodz** [comment]
I think the idea is to make the store behave as a regular JavaScript object, making it simpler to consume and interact with.

That way, developers don't need to do anything special while writing actions; simply access the store and mutate it normally. Any component using the store (i.e., an element with directives) will know when it needs to be re-rendered.
